### PR TITLE
Merging rafal branch with krzysiek and kuba branch

### PR DESCRIPTION
**Pull request by krzysieksr**

### CODE REVIEW

**Comment by gacikjakub** [comment]
Build 48 Still Failing
Deployed under: mergingRafalBranchWithKrzysiekAndKubaBranch.battleships.kuba005.nazwa.pl

**Review by agaped — APPROVED** [review]
In future you can add some javadocs.

File: backend/src/main/java/pl/krkteam/battleships/common/domain/ship/Ship.java [hunk 1]
```diff
@@ -24,15 +23,16 @@ public void attachMast(final Mast mast) {
     }
 
     public List<Mast> getMastList() {
-        return Collections.unmodifiableList(mastList);
+        return mastList;
```
Comment (boczkas) [inline backend/src/main/java/pl/krkteam/battleships/common/domain/ship/Ship.java]: unmodifiable

File: backend/src/main/java/pl/krkteam/battleships/common/dto/CoordinateDTO.java [hunk 2]
```diff
@@ -34,4 +31,4 @@ public int getX() {
     public int getY() {
         return rowIndex;
     }
-}
+}
```
Comment (boczkas) [inline backend/src/main/java/pl/krkteam/battleships/common/dto/CoordinateDTO.java]: \n

File: backend/src/main/java/pl/krkteam/battleships/shooting/dto/result/ResultWrongShotDTO.java [hunk 3]
```diff
@@ -9,7 +9,7 @@ public ResultWrongShotDTO() {
     }
 
     @Override
-    public OpponentShotResult getOpponentShotResult(int x, int y) {
+    public OpponentShotResult getOpponentShotResult(int y, int x) {
         return null;
```
Comment (boczkas) [inline backend/src/main/java/pl/krkteam/battleships/shooting/dto/result/ResultWrongShotDTO.java]: change null to something else

File: backend/src/main/java/pl/krkteam/battleships/turns/holding/TurnHolder.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+import pl.krkteam.battleships.common.domain.player.Player;
+import pl.krkteam.battleships.shooting.dto.result.ShotResultDTO;
+
+import java.util.NoSuchElementException;
+
+@Component
+public class TurnHolder {
+
+    private volatile Player currentPlayer;
+
```
Comment (boczkas) [inline backend/src/main/java/pl/krkteam/battleships/turns/holding/TurnHolder.java]: remove \n

File: backend/src/test/java/pl/krkteam/battleships/common/dto/CoordinateDTOTest.java [hunk 5]
```diff
@@ -8,7 +8,7 @@
 
     @Test
     public void testToString() {
-        CoordinateDTO coordinateDTO = new CoordinateDTO(2, 3);
+        CoordinateDTO coordinateDTO = new CoordinateDTO(3, 2);
```
Comment (boczkas) [inline backend/src/test/java/pl/krkteam/battleships/common/dto/CoordinateDTOTest.java]: extract it:
CoordinateDTO coordinateDTO = new CoordinateDTO(3, 2);

File: backend/src/test/java/pl/krkteam/battleships/common/dto/PlacingValidationResultDTOTest.java [hunk 6]
```diff
@@ -11,7 +11,7 @@ public void testGetResult() {
         PlacingValidationResultDTO placingValidationResultDTO = new PlacingValidationResultDTO(
                 PlacingValidationResultDTO.Result.OK);
 
-        assertEquals(placingValidationResultDTO.getResult(),PlacingValidationResultDTO.Result.OK);
+        assertEquals(placingValidationResultDTO.getResult(), PlacingValidationResultDTO.Result.OK);
 
     }
 }
```
Comment (boczkas) [inline backend/src/test/java/pl/krkteam/battleships/common/dto/PlacingValidationResultDTOTest.java]: \n

File: backend/src/test/java/pl/krkteam/battleships/ships/placing/validation/ShipControllerTest.java [hunk 7]
```diff
@@ -98,9 +98,9 @@ public void testValidateAndSaveShips() throws Exception {
...
                 .content(shipsJson)
-                .param("playerName", "SomePlayer")
-        )
-                .andExpect(status().is2xxSuccessful());
+                .param("playerName", "SomePlayer"))
+                .andExpect(status()
+                        .is2xxSuccessful());
     }
 
 }
```
Comment (boczkas) [inline backend/src/test/java/pl/krkteam/battleships/ships/placing/validation/ShipControllerTest.java]: \n

File: backend/src/test/java/pl/krkteam/battleships/ships/placing/validation/services/ShipRangeOnBoardValidatorTest.java [hunk 8]
```diff
@@ -15,113 +15,112 @@
...
     public void testIsCoordinateInBoardForRightBottomOutOfRangeX() {
-        CoordinatesFromJson coordinatesFromJson =new CoordinatesFromJson(3,10);
-        BoardSize boardSize=board.getBoardSize();
+        CoordinatesFromJson coordinatesFromJson = new CoordinatesFromJson(3, 10);
+        BoardSize boardSize = board.getBoardSize();
 
-        assertFalse(ShipRangeOnBoardValidator.isCoordinateInBoard(coordinatesFromJson,boardSize));
+        assertFalse(ShipRangeOnBoardValidator.isCoordinateInBoard(coordinatesFromJson, boardSize));
     }
 }
```
Comment (boczkas) [inline backend/src/test/java/pl/krkteam/battleships/ships/placing/validation/services/ShipRangeOnBoardValidatorTest.java]: \n

File: backend/src/test/java/pl/krkteam/battleships/shooting/controller/ShootingControllerTest.java [hunk 9]
```diff
@@ -36,10 +36,7 @@ public void setUp() {
 
     @Test
     public void testValidateShotAndExpectClientError() throws Exception {
-        mockMvc.perform(post("/game/player/shot")
-                .content("")
-                .param("playerName", "SomePlayer")
-        )
+        mockMvc.perform(post("/game/player/shot").content("").param("playerName", "SomePlayer"))
```
Comment (boczkas) [inline backend/src/test/java/pl/krkteam/battleships/shooting/controller/ShootingControllerTest.java]: self comment:
change it to separate lines

File: backend/src/test/java/pl/krkteam/battleships/turns.holding/TurnHolderTest.java [hunk 10]
```diff
@@ -0,0 +1,211 @@
...
+    @Mock
+    Player player1;
+
+    @Mock
+    Player player2;
+
+    @Mock
+    Player player3;
+
+    @BeforeClass
```
Comment (boczkas) [inline backend/src/test/java/pl/krkteam/battleships/turns.holding/TurnHolderTest.java]: do you need both annotations?

File: backend/src/test/java/pl/krkteam/battleships/turns.holding/TurnHolderTest.java [hunk 11]
```diff
@@ -0,0 +1,211 @@
...
+        turnHolder.addPlayer(playerOne);
+        turnHolder.addPlayer(playerTwo);
+        // when
+        turnHolder.choseFirst(playerTwo);
+        // then
+        Assert.assertEquals(turnHolder.getCurrentPlayer(), playerTwo);
+    }
+
+    @Test(dataProvider = "twoProperPlayers")
+    public void shouldChangeCurrentPlayerWhenHeMissed(Player playerOne, Player playerTwo) {
```
Comment (boczkas) [inline backend/src/test/java/pl/krkteam/battleships/turns.holding/TurnHolderTest.java]: WhenHeMissed -> WhenPlayerMissed/WhenMissed

**Review by boczkas — CHANGES_REQUESTED** [review]
cosmetic

**Review by boczkas — APPROVED** [review]
:+1:

**Review by RafalJDev — APPROVED** [review]
I was reviewing before pull request

**Pull request merged**